src/sentry/deletions/defaults/project.py

```diff
@@ -62,6 +56,12 @@ def get_child_relations(self, instance):
             ]
         )
         relations.append(ModelRelation(models.Group, {"project_id": instance.id}))
+        relations.append(
```

Comment:
Should we have a comment that explicitly calls out the ordering requirement?

Reply:
This PR explains why the relations are reordered.
Would a brief in-code comment also be useful for future devs.